### DISCUSSION TITLE
[Feature Request] Add regex support and organization level configuration for autolinks

### DISCUSSION THREAD

**Discussion by kishorviswanathan** [community/community#109128] [Enterprise]
My organization recently moved our repositories from Bitbucket to GitHub and we would like to enable autolinks to link pull requests to their corresponding issue in Jira. The current autolink configuration is really limited. It only allows specifying a static prefix but we would like to create links per project in Jira. That would require us to add 100s of autolinks, which is also fine if a configuration was allowed at the organization level. It is impossible to create 100s of autolinks in 100s of repositories manually.

Requested features:
1. Allow regular expression in autolinks.
2. Add option to configure autolinks at the organization level, which applies to all repositories.

**Comment by hfvr34** [comment]
We have a similar need: our tickets IDs are of a fixed length, but the prefix is a single character and so matches far too easily.

If we could specify either:
 • a more specific pattern using a regular expression (preferred), or
 • have a third match option of "only match digits and hyphens", or 
 • a more flexible character-class selector allowing to include/exclude different character classes from ticket ID matches
 
 … then we'd be able to implement something that worked for us.

**Reply by falkenbt** [reply]
We also struggle with this. As we create the repositories via HCP Terraform this is not really a technical problem but rather a financial one because each of the autolink resources is one resource in terraform. Considering about 100 JIRA spaces and 1000 repositories, we end up with 100K resources in HCP terraform, which is expensive. 

So both requested features, allowing regular expressions and especially autolinks at the org level would solve this problem. Until then we need to build a workaround to not manage the autolinks via Terraform, which adds complexity and technical debt.
